File: 프로그래머스/0/181829. 이차원 배열 대각선 순회하기/이차원 배열 대각선 순회하기.java
import java.util.stream.IntStream;
import java.util.Arrays;

class Solution {
    public int solution(int[][] board, int k) {
        return IntStream.range(0, board.length)
            .flatMap(i -> IntStream.range(0, board[i].length)
                .filter(j -> i + j <= k)
                .map(j -> board[i][j]))
            .sum();
    }
}

File: 프로그래머스/0/181829. 이차원 배열 대각선 순회하기/이차원 배열 대각선 순회하기.py
def solution(board, k):
    return sum([board[i][j] for i in range(0, min(k+1, len(board))) for j in range(0, min(k - i + 1, len(board[0])))])
print(solution([[0, 1, 2],[1, 2, 3],[2, 3, 4],[3, 4, 5]], 2))